### DISCUSSION TITLE
Improving SEO for Discussions and Issues

### DISCUSSION THREAD

**Discussion by cgarciae** [community/community#40281] [Discussions]
### Select Topic Area

Product Feedback

### Body

Hi Github!

On [Flax](https://github.com/google/flax) we've notice that SEO is a bit poor for our Issues and Discussions, for example if we search for google/flax#2609 verbatim by the title e.g. "[flax simplify submodule access from top-level Module](https://www.google.com/search?q=flax+simplify+submodule+access+from+top-level+Module&client=firefox-b-d&sxsrf=ALiCzsajxA6uUnCvNYsZeIlhBRqW95662A%3A1669650052944&ei=hNaEY7imOaucwbkP2q6MqAU&ved=0ahUKEwi4-LeSm9H7AhUrTjABHVoXA1UQ4dUDCA4&uact=5&oq=flax+simplify+submodule+access+from+top-level+Module&gs_lcp=Cgxnd3Mtd2l6LXNlcnAQAzIECCMQJzoKCAAQRxDWBBCwA0oECEEYAEoECEYYAFDFBVjyDGDkD2gBcAF4AIAB3AGIAdQEkgEFMC4zLjGYAQCgAQHIAQjAAQE&sclient=gws-wiz-serp)", Google cannot find it. We are wondering if there is something we can do or if its something Github could improve on? As you know Issues and Discussion can often serve as documentation so these being easily discovered is important for us.

Thanks!

**Comment by ossanna16** [comment]
Hi there @cgarciae and welcome to our community! Thank you for the feedback!

**Comment by entcheva** [comment]
Hi @cgarciae, thank you for the feedback. I'm an Engineering Manager on the Discussions team 👋 We've done some work in the past to improve discussions SEO, but have recently received feedback similar to yours that suggests we have more work to do here to improve SEO. I've added your feedback to our existing internal issue and will continue to work with our Discussions leads to prioritize and move this forward.

**Reply by xamidi** [reply]
Is there any progress? I still have this issue. For example, when googling for `CCpCCNpqrCsCCNtCrtCpt`, I would wish for [a related discussion](https://github.com/xamidi/pmGenerator/discussions/10) to show up, but Google seems totally unaware of that site. It works as desired for repository READMEs.
**1st Edit:** Now, oddly, it also works for this discussion, *but not for my own*.
**2nd Edit:** Google now indexed it (one month after creation)! Not sure why, though.
**3rd Edit:** Changed link.

**Comment by Cockfight2** [comment]
Publish Relevant, Authoritative Content .
Update Your Content Regularly .
Metadata . 
Have a link-worthy site .
Use alt tags [.](https://gamefowlforsale.com/)

**Comment by davanstrien** [comment]
Apologies for replying to an old thread. I'd like to know if there has been much work on this topic. Big fan of discussions, but Google searches often struggle to surface relevant content from discussion threads even when using exact search for discussion titles, etc.

**Comment by clareqinty89** [comment]
SEO for discussions and issues can be a real pain sometimes. It's crucial to have that stuff easily discoverable, especially when it serves as documentation.
As for a fix, there are a few things you could try. Make sure your titles are descriptive and include relevant keywords. Also, consider linkin' to those discussions from other high-traffic pages—it can give 'em a boost in visibility.
Oh, and here's a little trick I've picked up: CTR boosting. Gettin' those clicks rollin' in can really amp up your SEO game, trust me.
And hey, if you're still searchin' for more tips, I recently stumbled upon [searchseo.io](https://www.searchseo.io/ctr-manipulation)—it's packed with solid SEO advice that might just help you crack this nut.

**Comment by Xelliese** [comment]
To start improving your company's online presence through SEO, I'd recommend focusing on a few key strategies. First, make sure you're targeting the right keywords that your audience is searching for. You can use tools like Google Keyword Planner or Ahrefs to identify these keywords and then optimize your website content around them. Additionally, regularly updating your content with fresh, relevant information helps improve your rankings. Building quality backlinks and improving your website's loading speed are also crucial factors that can boost your search engine rankings.

**Comment by AldousHuxley786** [comment]
Improving SEO for discussions and issues means optimizing keywords, adding relevant tags, and ensuring content is easily searchable. In my [SEO agency](https://pearllemonseo.ca/), we focus on these tactics to boost visibility and engagement!

**Comment by lucatosc** [comment]
To improve SEO for your GitHub Issues and Discussions, you can try the following:

1. **Use Clear Titles**: Ensure that your issue and discussion titles are concise and descriptive. Including keywords that users might search for can help.

2. **Enhance Descriptions**: Provide detailed descriptions with relevant keywords and phrases. This can improve indexing by search engines.

3. **FAQs and Tags**: Create Frequently Asked Questions or tags within your discussions to categorize topics and improve searchability.

4. **Linking**: Link to related issues and discussions within your content to create a web of interconnected information.

5. **External Linking**: Share links to relevant issues and discussions on social media, blogs, or other platforms to increase visibility.

6. **Feedback to GitHub**: Consider submitting feedback to GitHub about enhancing SEO for Issues and Discussions, as they have control over site indexing.

By implementing these strategies, you can enhance the discoverability of your GitHub content.

**Comment by bartmichalak** [comment]
@entcheva Just wondering if there's any update on this topic? It seems like many threads aren't even indexed, even from popular repositories like https://github.com/rails/rails

Would love to hear more about it!
